File: java/src/main/java/com/alxarafe/app/application/movement/StockOperationException.java
package com.alxarafe.app.application.movement;

/**
 * Stock operation error carrying the HTTP status it maps to, so the HTTP
 * adapter translates it directly (404 unknown, 400 input semantics,
 * 409 warehouse state conflict).
 */
public final class StockOperationException extends RuntimeException {

    private final int status;

    public StockOperationException(String message, int status) {
        super(message);
        this.status = status;
    }

    public int status() {
        return status;
    }
}

File: php/src/Application/Movement/StockOperationException.php
<?php

declare(strict_types=1);

namespace Alxarafe\App\Application\Movement;

use RuntimeException;

/**
 * Error de operación de stock con su estado HTTP asociado, para que el
 * adaptador HTTP lo traduzca directamente (404 desconocido, 400 semántico
 * de entrada, 409 conflicto de estado del almacén).
 */
final class StockOperationException extends RuntimeException
{
    public function __construct(string $message, private int $status = 409)
    {
        parent::__construct($message);
    }

    public function status(): int
    {
        return $this->status;
    }
}
